src/MatrixRoomHandler.ts

```diff
@@ -25,9 +25,32 @@ const log = Logging.get("MatrixRoomHandler");
 
 const ACCOUNT_LOCK_MS = 1000;
 const EVENT_MAPPING_SIZE = 16384;
+const XEP0106_ESCAPING_TRANSFORMATIONS = {
+   "\"": "\22",
+   "&": "\26",
+   "'": "\27",
+   "/": "\2f",
+   ":": "\3a",
+   "<": "\3c",
+   ">": "\3e",
+   "@": "\40"
+}
+function XEP0106_APPLY_ESCAPING_TRANSFORMATIONS(roomId: string): string {
+    return roomId.split("").map(i => {
+        if(Object.keys(XEP0106_ESCAPING_TRANSFORMATIONS).includes(i))
```

Comment:
You can replace these 4 lines with `XEP0106_ESCAPING_TRANSFORMATIONS[char] || i`

No need to lookup beforehand.